internal/controllers/machinedeployment/machinedeployment_sync_test.go

```diff
@@ -495,6 +499,234 @@ func TestSyncDeploymentStatus(t *testing.T) {
 	}
 }
 
+func TestComputeDesiredMachineSet(t *testing.T) {
```

Comment:
I wonder if a table test is really the right format. It's very hard to match the various structs we build with the table test cases and then figure out if everything behaves as expected.

I think it would be easier to define some common structs and then just write down the test cases sequentially with all the assertions.
(see the way we did it in desired_state_test.go, e.g. TestComputeInfrastructureCluster)

@fabriziopandini WDYT?


(same for TestComputeDesiredMachine)

Reply:
I don't have objections, if we can make tests more easily maintanables I'm +1 for me given that we have the same end coverage at the end